### ISSUE TITLE
Baking plugin models can clash with existing app models

### ISSUE THREAD

**Issue by ndm2** [cakephp/bake#111]
The model task will use the model name without the plugin name to retrieve the table object, see
- https://github.com/cakephp/bake/blob/1.0.9/src/Shell/Task/ModelTask.php#L107
- https://github.com/cakephp/bake/blob/1.0.9/src/Shell/Task/ModelTask.php#L323
- https://github.com/cakephp/bake/blob/1.0.9/src/Shell/Task/ModelTask.php#L654

This leads to possible existing app table classes with same name being used, which could then in turn cause various problems, like for example the wrong accessible fields being generated as associations are not going to be applied, or problems with finding associations or counter cache fields as the schema could possible have been modified.

**Comment by markstory** [comment]
So you have models in a plugin and the app that are different, but use the same table? Or is this a problem when you have multiple databases with tables that share a name?

**Comment by ndm2** [comment]
Not necessarily the same db table, that might be actually a different one specified via the `--table` option, but the same classname, say `UsersTable`, then baking `Plugin.UsersTable` will use the apps `UsersTable`.

Here's a very, very ugly test to show how this could go wrong

```php
    public function testBakeModelPluginClash()
    {
        TableRegistry::get('PluginClash', [
            'className' => '\Bake\Test\App\Model\Table\ArticlesTable',
            'table' => 'bake_articles'
        ]);

        $this->Task->connection = 'test';

        $entityFile = $this->_normalizePath(APP . 'Plugin/ModelTest/src/Model/Entity/PluginClash.php');
        $this->Task->expects($this->at(1))
            ->method('createFile')
            ->with($entityFile, $this->stringContains('protected $_accessible = [
        \'bake_article_id\' => true,
        \'bake_user_id\' => true,
        \'comment\' => true,
        \'published\' => true,
        \'bake_article\' => true,
        \'bake_user\' => true,
    ];'));

        $this->Task->params['table'] = 'bake_comments';
        $this->Task->main('ModelTest.PluginClash');
    }
```

**Comment by josegonzalez** [comment]
Closed by #303.
